packages/flutter_tools/gradle/flutter.gradle

```diff
@@ -173,17 +173,20 @@ class FlutterPlugin implements Plugin<Project> {
                     matchingFallbacks = ["debug", "release"]
                 }
             }
-            release {
-                // Enables code shrinking, obfuscation, and optimization for only
-                // your project's release build type.
-                minifyEnabled true
-                // Enables resource shrinking, which is performed by the
-                // Android Gradle plugin.
-                // NOTE: The resource shrinker can't be used for libraries.
-                shrinkResources isBuiltAsApp(project)
-                // Fallback to `android/app/proguard-rules.pro`.
-                // This way, custom Proguard rules can be configured as needed.
-                proguardFiles project.android.getDefaultProguardFile("proguard-android.txt"), flutterProguardRules, "proguard-rules.pro"
+            // Shrinking is only false for multi apk split aot builds, where shrinking is not allowed.
```

Comment:
b/120517460 seems to also have some relevant info on this.

Reply:
Oh interesting! Seems a bit like a 2 steps forward 1 step back type of thing apk size wise :D. Though there isn't anything we can do. We should definitely document this thoroughly, since there may be security implications for add-to-Flutter user who add their own Java libraries.

Reply:
Since the experimental version is already out, this is definitely something can remove in the future. It should be in future releases of Gradle 4.2+. We can pitch this as a beta/preview/etc version of the deferred components for now if that seems appropriate due to this limitation